front/Dockerfile

```diff
@@ -1,23 +1,35 @@
 FROM node:14.18.2-buster-slim@sha256:20bedf0c09de887379e59a41c04284974f5fb529cf0e13aab613473ce298da3d as builder
-WORKDIR /usr/src
+
+WORKDIR /usr/src/messages
 COPY messages .
 RUN yarn install && yarn ts-proto
 
-# we are rebuilding on each deploy to cope with the PUSHER_URL environment URL
-FROM thecodingmachine/nodejs:14-apache
+WORKDIR /usr/src/front
+COPY front .
+
+# move messages to front
+RUN cp -r ../messages/ts-proto-generated/protos/* src/Messages/ts-proto-generated
+RUN sed -i 's/import { Observable } from "rxjs";/import type { Observable } from "rxjs";/g' src/Messages/ts-proto-generated/messages.ts
+RUN cp -r ../messages/JsonMessages src/Messages/JsonMessages
 
-COPY --chown=docker:docker front .
-COPY --from=builder --chown=docker:docker /usr/src/ts-proto-generated/protos /var/www/html/src/Messages/ts-proto-generated
-RUN sed -i 's/import { Observable } from "rxjs";/import type { Observable } from "rxjs";/g' /var/www/html/src/Messages/ts-proto-generated/messages.ts
-COPY --from=builder --chown=docker:docker /usr/src/JsonMessages /var/www/html/src/Messages/JsonMessages
+RUN yarn install && yarn build
 
 # Removing the iframe.html file from the final image as this adds a XSS attack.
 # iframe.html is only in dev mode to circumvent a limitation
 RUN rm dist/iframe.html
 
-RUN yarn install
+FROM thecodingmachine/nodejs:14-apache
+
+COPY --from=builder --chown=docker:docker /usr/src/front/dist dist
+COPY front/templater.sh .
+
+USER root
+RUN DEBIAN_FRONTEND=noninteractive apt-get update \
+    && apt-get install -y \
+    gettext-base \
```

Comment:
Is this the package containing `envsubst`?

Reply:
Yes